### PR TITLE
Resolve recent test failures

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR resolves recent test failures.

### CODE REVIEW

File: skyportal/models/user_notification.py [hunk 1]
```diff
@@ -402,7 +402,11 @@ def receive_after_flush(session, context):
 
         for user in users:
             # Only notify users who have read access to the new record in question
-            pref = user.preferences['notifications']
+            if user.preferences is not None:
+                pref = user.preferences['notifications']
+            else:
+                pref = None
```
Comment (guynir42) [inline skyportal/models/user_notification.py:408]: If you do `pref={}` you may be able to simplify the following conditionals.

File: skyportal/models/user_notification.py [hunk 2]
```diff
@@ -412,7 +416,9 @@ def receive_after_flush(session, context):
                 is not None
             ):
                 if is_gcnevent:
-                    if "gcn_notice_types" in pref["gcn_events"].keys():
+                    if (pref is not None) and "gcn_notice_types" in pref[
+                        "gcn_events"
+                    ].keys():
```
Comment (guynir42) [inline skyportal/models/user_notification.py:421]: Can remove `.keys()`

**Review by guynir42 — APPROVED** [review]
Minor comments below.

**Comment by guynir42** [comment]
I've restarted the failed test run. If it isn't all green I would spend a little more time investigating why before merging.

**Comment by mcoughlin** [comment]
Down to a test that works very well locally but struggles. Will spend more time this week.

**Pull request merged**